src/HDF5.jl

```diff
@@ -73,6 +73,13 @@ const Hssize  = Int64
 const Htri    = Cint   # pseudo-boolean (negative if error)
 const Haddr   = UInt64
 
+# MPI communicators required by H5P
+abstract  type Hmpih end
+primitive type Hmpih32 <: Hmpih 32 end # MPICH C/Fortran, OpenMPI Fortran: 32 bit handles
+primitive type Hmpih64 <: Hmpih 64 end # OpenMPI C: pointers (mostly 64 bit)
+const mpihandles = Dict(4 => Hmpih32, 8 => Hmpih64)
+h5_mpihandle(handle) = reinterpret(mpihandles[sizeof(handle)],handle)
```

Comment:
should the input here be  `handle::Integer`  ?

Reply:
This depends on the caller. In case of MPI.jl the handle will be a 4/8 byte `primitive type Ccomm` for the C communicator either wrapping a 4 byte integer handle (MPICH) or a 4/8 byte pointer (OpenMPI). I wanted to have this as generic as possible. This way users do not even have to use MPI.jl, but can just pass the pointers/handles without conversion.